indico/modules/designer/util.py

```diff
@@ -98,3 +104,55 @@ def get_badge_format(tpl):
     return next((frm for frm, frm_size in format_map.items()
                  if (frm_size[0] == float(tpl.data['width']) / PIXELS_CM and
                      frm_size[1] == float(tpl.data['height']) / PIXELS_CM)), 'custom')
+
+
+def has_regform_field_placeholders(data):
+    """Return `True` if the given template data contains regform field placeholders."""
+    return any(is_regform_field_placeholder(item) for item in data['items'])
+
+
+def remove_regform_field_placeholders(data):
+    """Return a copy of the template data with regform field placeholders removed."""
+    return data | {
+        'items': [item for item in data['items'] if not is_regform_field_placeholder(item)]
+    }
+
+
+def can_link_to_regform(template, regform):
+    """Return True if the given template can be linked to the given registration form."""
+    return regform in get_linkable_regforms(template)
+
+
+def get_linkable_regforms(template):
+    """Return all registration forms that can be linked to the given template."""
+    if template.category or template.type == TemplateType.poster or template.registration_form:
+        return []
+
+    query = RegistrationForm.query.with_parent(template.event).filter(
+        ~RegistrationForm.is_deleted,
+    )
+
+    return [
+        regform for regform in query
+        if (  # If the template has a backside, the backside must be linked to the same regform
+            template.backside_template is None
+            or template.backside_template.registration_form is None
+            or regform == template.backside_template.registration_form
+        )
+        and all(  # If the the template is a backside, all frontsides must be linked to the same regform
+            tpl.registration_form is None or tpl.registration_form == regform for tpl in template.backside_template_of
+        )
+    ]
+
+
+def get_printable_event_templates(regform):
+    """Return all templates that can be used to print badges/tickets for the given registration form.
+
+    A template can be used if it is either not linked to anything or linked to the current registation form.
+    """
+    return DesignerTemplate.query.with_parent(regform.event).filter(
+        db.or_(
+            DesignerTemplate.registration_form_id.is_(None),
+            DesignerTemplate.registration_form == regform
+        )
+    ).all()
```

Comment:
Our usual formatting for multiline queries

```suggestion
    return (
        DesignerTemplate.query
        .with_parent(regform.event)
        .filter(db.or_(DesignerTemplate.registration_form_id.is_(None),
                       DesignerTemplate.registration_form == regform))
        .all()
    )
```